### ISSUE TITLE
getImageURI()	 is not a function

### ISSUE THREAD

**Issue by nickypeh** [angular-google-chart/angular-google-chart#261]
I try to access the chartWrapper.getChartType().getImageURI() at the agcOnReady as the function pass in ChartType is different. 

However, it throw out a error saying getImageURI() is not a function. 
What could have caused this as I see the chartobject should have this function in the google chart documentation. 

Please help.

**Comment by nbering** [comment]
`chartWrapper.getChartType()` probably returns a string or constructor, I'd have to look that up. To get the chart instance you need `chartWrapper.getChart()`.

**Comment by nickypeh** [comment]
Sorry for that, what i tried is the chartWrapper.getChart() instead of the getChartType. Copied the wrong function. However, I tried the getChart() but it still return getImageURI() is not a function. 
Below is the codes I have tried on. 
` vm.barChart = {
                type: 'Bar',
                data: [
                    ['Year', 'Sales', 'Expenses', 'Profit'],
                    ['2014', 1000, 400, 200],
                    ['2015', 1170, 460, 250],
                    ['2016', 660, 1120, 300],
                    ['2017', 1030, 540, 350]
                ],
                options: {
                    chart: {
                        title: 'Company Performance',
                        subtitle: 'Sales, Expenses, and Profit: 2014-2017'
                    },
                    bars: 'vertical',
                    width: 800,
                    height: 600
                }
            };
           vm.readyHandler =  readyHandler;
            function readyHandler(chartWrapper){
                var chart = chartWrapper.getChart();
                var img = chart.getImageURI();
            }`

> For the HTML `<div google-chart chart="vm.barChart" agc-on-ready="vm.readyHandler(chartWrapper)"></div> `

**Comment by nbering** [comment]
Sorry, Google's made a mess of the documentation for Bar Charts in my opinion. There are two types of Bar Charts availables, they "classic" charts version of the "material" version.

The classic charts version should have a `type: 'BarChart'` and the material version is loaded from a non-default package and can be used with `type: 'charts.Bar'`. You can find more on that on my blog post: [Angular-Google-Chart with Material Design Bar Charts](http://nicholasbering.ca/angularjs/2015/02/10/agc-material-bar/).

I'm not sure which it is your intent to use, but the Material charts are - to my knowledge - not yet at feature parity with classic charts, so I would be entirely unsurprised if they did not have this function, yet.

Try setting a breakpoint in your function and see what sort of object you are actually getting from chartWrapper.getChart() to see if you're getting a valid chart at all. The ready handler gets called every time the ChartWrapper instance is finished doing something and is ready to accept commands. You might have to check if the chart returned is valid, if not it may be been calling ready because it was prepared to accept requests after instantiation, and then catch the ready that is called when it is finished drawing.

**Comment by nickypeh** [comment]
Thanks for the information. It helps! :+1: 
Guess the material charts do not have the getImageURI function yet. 
Any idea they have full documentation for the google material charts ? I could not find any documentation regarding the chart and available functions.

**Comment by nbering** [comment]
I agree that the documentation on Bar Charts has become somewhat fragmented since the Google Visualization Team started pushing them toward being the standard. After responding to your initial issue I raised an issue about that at google/google-visualization-issues#2214. Feel free to add your own comments to that issue. I'm pretty sure they're kind of ignoring me, possibly because I'm so active in their issues repository. I've made other comments about documentation and they haven't responded, so if someone else could add their voice it might help.
